dds/DCPS/transport/rtps_udp/RoutedGuidPair.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#ifndef OPENDDS_DCPS_TRANSPORT_RTPS_UDP_ROUTEDGUIDPAIR_H
+#define OPENDDS_DCPS_TRANSPORT_RTPS_UDP_ROUTEDGUIDPAIR_H
+
+#include "Rtps_Udp_Export.h"
+#include <dds/DCPS/GuidUtils.h>
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace DCPS {
+
+#pragma pack(push, 1)
+struct OpenDDS_Rtps_Udp_Export RoutedGuidPair {
+  RoutedGuidPair(const GUID_t& src, const GUID_t dst) : src_(src), dst_(dst) {}
+  bool operator<(const RoutedGuidPair& rhs) const {
+    return std::memcmp(static_cast<const void*>(&src_), static_cast<const void*>(&rhs.src_), 2 * sizeof(GUID_t)) < 0;
+  }
+  bool operator==(const RoutedGuidPair& rhs) const {
+    return std::memcmp(static_cast<const void*>(&src_), static_cast<const void*>(&rhs.src_), 2 * sizeof(GUID_t)) == 0;
+  }
+  GUID_t src_;
```

Comment:
Add a comment that the comparison functions assume certain things about the memory layout so don't rearrange members without updating the comparison functions.